scripts/demo_assets/static/index.js

```diff
@@ -17,22 +49,86 @@ function load_wasm() {
   // Here we tell bindgen the path to the wasm file so it can start
   // initialization and return to us a promise when it's done.
 
-  console.debug("loading wasm…");
-  wasm_bindgen("re_viewer_bg.wasm").then(on_wasm_loaded).catch(on_wasm_error);
+  document.getElementById("center_text").innerHTML = `
+  <p class="strong">
+      Loading Application Bundle...
+  </p>
+  <p class="subdued" id="status">
+  </p>`;
+
+  const status_element = document.getElementById('status');
+  function progress({ loaded, total_bytes }) {
+    if (total_bytes != null) {
+      status_element.innerHTML = Math.round(Math.min(loaded / total_bytes * 100, 100)) + '%';
+    } else {
+      status_element.innerHTML = (loaded / (1024 * 1024)).toFixed(1) + 'MiB'
+    }
+  }
+
+  var timeoutId = setTimeout(function () {
+    document.getElementById("center_text").classList.remove("hidden");
+    document.getElementById("center_text").classList.add("visible");
+  }, 1500);
+
+  async function wasm_with_progress() {
+    const response = await fetch('./re_viewer_bg.wasm');
+    // Use the uncompressed size
+    var content_length;
+    var content_multiplier = 1;
+    // If the content is gzip encoded, try to get the uncompressed size.
+    if (response.headers.get('content-encoding') == 'gzip') {
```

Comment:
why just `gzip`, and not `compress`, `deflate`, br`, etc?

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding

Reply:
gzip is what gcloud happens to do. I'll add support for more encodings in the future:
https://github.com/rerun-io/rerun/issues/2452